source/OrderedTypes/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,875 @@
+  open Joins (rel-syntax 2-Poset)
+  open δ-complete-poset 𝓥 2-Poset
+  open non-trivial-posets 2-Poset  
+
+  sup-from-δ-completeness : Σ s ꞉ ∣ 2-Poset ∣ₚ ,
+                          (s is-lub-of (δ-fam ₀ ₁ (P , P-is-prop))) holds
+  sup-from-δ-completeness = i ₀ ₁ ⋆ (P , P-is-prop)
+
+  sup-gives-wem : Σ s ꞉ ∣ 2-Poset ∣ₚ ,
+                           (s is-lub-of (δ-fam ₀ ₁ (P , P-is-prop))) holds
+                         → ¬ P + ¬ (¬ P)
+  sup-gives-wem (₀ , sup) =
+    inl (pr₁ (WEM-lemma 𝓥 2-is-non-trivial (P , P-is-prop)) sup)
+  sup-gives-wem (₁ , sup) =
+    inr (pr₂ (WEM-lemma 𝓥 2-is-non-trivial (P , P-is-prop)) sup)
+
+  wem : ¬ P + ¬ (¬ P)
+  wem = sup-gives-wem sup-from-δ-completeness
+
+\end{code}
+
+Since non-trivial is a negated concept it only has enough strength to derive
+WEM, we now introduce the stronger concept of positivity.
+
+\begin{code}
+
+module Positive-Posets (𝓤  𝓦  𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+ 
+ open δ-complete-poset 𝓥 A
+ open Universal fe
+ open PosetReasoning A
+ open Joins (_≤_)
+
+ module positive-posets (i : is-δ-complete) where
+
+  _<_ : (x y : ∣ A ∣ₚ) → 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+  x < y = (x ≤ y) holds
+        × ((z : ∣ A ∣ₚ)
+          → (y ≤ z) holds
+          → (P : Ω 𝓥)
+          → (z is-lub-of (δ-fam x z P)) holds
+          → P holds)
+
+  order-from-strictly-below : {x y : ∣ A ∣ₚ} → x < y → (x ≤ y) holds
+  order-from-strictly-below c = pr₁ c
+
+  sup-condition-from-strictly-below : {x y : ∣ A ∣ₚ}
+                                    → x < y
+                                    → ((z : ∣ A ∣ₚ)
+                                     → (y ≤ z) holds
+                                     → (P : Ω 𝓥)
+                                     → (z is-lub-of (δ-fam x z P)) holds
+                                     → P holds)
+  sup-condition-from-strictly-below x-strictly-below-y =
+     pr₂ x-strictly-below-y
+
+  strictly-below-implies-non-trivial : (x y : ∣ A ∣ₚ)
+                                     → is-δ-complete
+                                     → (x < y)
+                                     → (x ≤ y) holds × (x ≠ y)
+  pr₁ (strictly-below-implies-non-trivial x y i x-strictly-below-y) =
+     order-from-strictly-below x-strictly-below-y
+  pr₂ (strictly-below-implies-non-trivial x y i x-strictly-below-y) p =
+     𝟘-elim (sup-condition-from-strictly-below x-strictly-below-y y
+                                               (≤-is-reflexive A y) ⊥
+                                               (y-is-ub , y-has-lub-cond))
+    where
+     y-is-ub : (y is-an-upper-bound-of (δ-fam x y ⊥)) holds
+     y-is-ub (inl ⋆) = order-from-strictly-below x-strictly-below-y
+
+     y-has-lub-cond : ((u , _) : upper-bound (δ-fam x y ⊥)) → (y ≤ u) holds
+     y-has-lub-cond (u , is-upbnd) = y ＝⟨ p ⁻¹ ⟩ₚ is-upbnd (inl ⋆)
+
+\end{code}
+
+TODO: We could show that if the converse holds then so does EM in 𝓥.
+This is because in particular, for x,y : Ω 𝓥
+
+  if x ≤ y and x ≠ y implies x < y then EM holds in 𝓥
```

Comment:
It would be helpful to say here that we get DNE (not-not P -> P), which is equivalent to EM, by taking x = \bot and y = P.